packages/amplify-util-uibuilder/src/utils/prePushHandler.ts

```diff
@@ -0,0 +1,55 @@
+import { $TSContext } from 'amplify-cli-core';
+import type { Schema } from '@aws-amplify/datastore';
+import { printer } from 'amplify-prompts';
+import AmplifyUIBuilder from 'aws-sdk/clients/amplifyuibuilder';
+import { AmplifyStudioClient } from '../clients';
+import { isFormDetachedFromModel, isFormSchemaCustomized, shouldRenderComponents } from '../commands/utils';
+
+/**
+ * Handles showing a warning for detached forms pre-push.
+ */
+export const prePushHandler = async (context: $TSContext): Promise<void> => {
+  if (!(await shouldRenderComponents(context))) {
+    return;
+  }
+  try {
```

Comment:
This try/catch seems redundant with the caller's behavior: https://github.com/aws-amplify/amplify-cli/pull/11610/files/3be18a90df5c548545f8cf96c58faee8d722a1f6#diff-29aed445c842f7a58104e2fc95802749904534fe0680381ce758b1536a34eb36R10